packages/cli/yaml/docs-validator/src/rules/valid-markdown-link/check-if-pathname-exists.ts

```diff
@@ -0,0 +1,110 @@
+import { wrapWithHttps } from "@fern-api/docs-resolver";
+import { AbsoluteFilePath, RelativeFilePath, dirname, doesPathExist, join } from "@fern-api/fs-utils";
+
+import { getRedirectForPath } from "./redirect-for-path";
+import { addLeadingSlash, removeLeadingSlash } from "./url-utils";
+
+/**
+ * Checks if the given path exists in the docs.
+ *
+ * For a given pathname, there's two scenarios:
+ * 1. the pathname is a slug to a page in the docs (e.g. `/docs/my-page`)
+ * 2. the pathname is a path to a file in the docs (e.g. `/docs/my-file.md`)
+ *
+ * Crucially, the current file may referenced by two slugs, so if the target pathname contains a relativized path,
+ * we will need to check that if the relativized paths can be found on all of the slugs for the current file.
+ */
+export async function checkIfPathnameExists(
+    pathname: string,
```

Comment:
nit: Why not include the `pathname` arg alongside the rest of the in-lined argument struct? That way _every_ parameter is named rather than an anonymous parameter to begin.

I get what you're going for with the function name (i.e. the pathname is the thing I'm acting upon), but I personally like using the named struct for everything, especially for consistency with the rest of the codebase.

Feel free to take it or leave it.

Reply:
Nah I think you're right